rudderstack/integrations/destinations/common_config_meta_test.go

```diff
@@ -0,0 +1,199 @@
+	}{
+		{
+			description:          "Valid list of supported source types",
+			supportedSourceTypes: []string{"web", "android", "ios"},
+			expectedProperties: []c.ConfigProperty{
+				c.ArrayWithStrings("oneTrustCookieCategories.web", "oneTrustCookieCategory", "onetrust_cookie_categories.0.web"),
+				c.ArrayWithStrings("oneTrustCookieCategories.android", "oneTrustCookieCategory", "onetrust_cookie_categories.0.android"),
+				c.ArrayWithStrings("oneTrustCookieCategories.ios", "oneTrustCookieCategory", "onetrust_cookie_categories.0.ios"),
+			},
+			expectedSchema: map[string]*schema.Schema{
+				"onetrust_cookie_categories": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Description: "Specify OneTrust category IDs.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"web": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								Elem:        &schema.Schema{Type: schema.TypeString},
+							},
+							"android": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								Elem:        &schema.Schema{Type: schema.TypeString},
+							},
+							"ios": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								Elem:        &schema.Schema{Type: schema.TypeString},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			description:          "Empty list of supported source types",
+			supportedSourceTypes: []string{},
+			expectedProperties:   []c.ConfigProperty{},
+			expectedSchema:       map[string]*schema.Schema{
+				"onetrust_cookie_categories": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Description: "Specify OneTrust category IDs.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{},
+					},
+				},
+			},
+		},
+		{
+			description:          "A single supported source type",
+			supportedSourceTypes: []string{"web"},
+			expectedProperties: []c.ConfigProperty{
+				c.ArrayWithStrings("oneTrustCookieCategories.web", "oneTrustCookieCategory", "onetrust_cookie_categories.0.web"),
+			},
+			expectedSchema: map[string]*schema.Schema{
+				"onetrust_cookie_categories": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Description: "Specify OneTrust category IDs.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"web": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								Elem:        &schema.Schema{Type: schema.TypeString},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			actualProperties, actualSchema := GetCommonConfigMeta(tc.supportedSourceTypes)
+			require.EqualValues(t, tc.expectedSchema, actualSchema)
+			require.True(t, len(actualProperties) == len(tc.expectedProperties))
+		})
+	}
+}
+
+func TestGetConfigMetaForOneTrustConsents(t *testing.T) {
```

Comment:
I feel these tests are redundant since they are mostly testing part of the common config which is already tested in the previous test. In practice, `TestGetConfigMetaForOneTrustConsents` is an internal implementation which is never called directly by destination implementations.

Reply:
Addressed.